### PR TITLE
T17 - Permutations

### PR DESCRIPTION
**Pull request by Ajay07Metamation**

### CODE REVIEW

File: Training/Program.cs [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
-        }
-    }
+﻿//Program to get all possible permutations of a string
+using static System.Console;
+string answer = "";
+Write ("Enter the string : ");
+string word = ReadLine () ?? "";
+if (word.Length > 0 && word.All (char.IsLetter)) {
+   Write ("\nAll Possible Permutations are : \n");
+   GetPermute (word, answer);
```
Comment (MetaNaveen) [inline Training/Program.cs:8]: The method name implies that something should be returned. The name could have been "PrintPermutation ()".

Anyways, could you get array of permuted words as result of GetPermute() method and the print it?

File: Training/Program.cs [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+} else WriteLine ("Enter Valid Input");
+void GetPermute (string word, string answer) {
```
Comment (MetaNaveen) [inline Training/Program.cs:10]: We use an empty line before a Method definition as a coding style guideline.

**Review by MetaNaveen — CHANGES_REQUESTED** [review]

File: Training/Program.cs (hunk 2 above)
Comment (Ajay07Metamation) [reply Training/Program.cs]: All changes and formatting done in latest commit

**Comment by MetaNaveen** [comment]
**Final Notes For Consideration:**

- Simplify using var on LHS or new () on RHS.
- It now says 'PrintPermute' but only computes the permutation to store it in the caller's List object. Name appropriately.

**Pull request closed without merging**